openedx_learning/apps/authoring/containers/api.py

```diff
@@ -0,0 +1,380 @@
+
+    Returns:
+        The newly created entity list rows.
+    """
+    order_nums = range(len(entity_pks))
+    with atomic():
+        entity_list_rows = EntityListRow.objects.bulk_create(
+            [
+                EntityListRow(
+                    entity_list=entity_list,
+                    entity_id=entity_pk,
+                    order_num=order_num,
+                    draft_version_id=draft_version_pk,
+                    published_version_id=published_version_pk,
+                )
+                for entity_pk, order_num, draft_version_pk, published_version_pk in zip(
+                    entity_pks, order_nums, draft_version_pks, published_version_pks
+                )
+            ]
+        )
+
+    return entity_list_rows
+
+
+def create_entity_list_with_rows(
+    entity_pks: list[int],
+    draft_version_pks: list[int | None],
+    published_version_pks: list[int | None],
+) -> EntityList:
+    """
+    Create a new entity list with rows.
+
+    Args:
+        entity_pks: The IDs of the publishable entities that the entity list rows
+        reference.
+        order_nums: The order_nums of the entity list rows in the entity list.
+        draft_version_pks: The IDs of the draft versions of the entities
+        (PublishableEntityVersion) that the entity list rows reference.
+        published_version_pks: The IDs of the published versions of the entities
+        (PublishableEntityVersion) that the entity list rows reference.
+
+    Returns:
+        The newly created entity list.
+    """
+    entity_list = create_entity_list()
+    create_entity_list_rows(
+        entity_list=entity_list,
+        entity_pks=entity_pks,
+        draft_version_pks=draft_version_pks,
+        published_version_pks=published_version_pks,
+    )
+    return entity_list
+
+
+def copy_rows_to_new_entity_list(
+    rows: QuerySet[EntityListRow],
+) -> EntityList:
+    """
+    Copy rows from an existing entity list to a new entity list.
+
+    Args:
+        entity_list: The entity list to copy the rows to.
+        rows: The rows to copy to the new entity list.
+
+    Returns:
+        The newly created entity list.
+    """
+    entity_list = create_entity_list()
+    with atomic():
+        entity_list_rows = EntityListRow.objects.bulk_create(
+            [
+                EntityListRow(
+                    entity_list=entity_list,
+                    entity_id=row.entity.id,
+                    order_num=row.order_num,
+                    draft_version_id=row.entity.draft.version.pk,
+                    published_version_id=row.entity.published.version.pk,
+                )
+                for row in rows
+            ]
+        )
+
+    return entity_list
+
+
+def create_container_version(
+    container_pk: int,
+    version_num: int,
+    title: str,
+    publishable_entities_pk: list[int],
+    draft_version_pks: list[int | None],
+    published_version_pks: list[int | None],
```

Comment:
Should the draft/published versions be derived somehow from the publishable entities?